src/utils/price.ts

```diff
@@ -0,0 +1,39 @@
+import BigNumber from 'bignumber.js'
+import BN from 'bn.js'
+
+interface Token {
+  amount: BN | BigNumber | string
+  decimals?: number
+}
+
+interface CalculatePriceParams {
+  buyToken: Token
+  sellToken: Token
+}
+
+/**
+ * Calculates price from either BN, BigNumber or string amounts
+ * based on decimal precision of each token.
+ *
+ * Decimals defaults to 0.
+ * Use case is to calculate the price when values already in the same unit
+ * and no adjustment is required.
+ *
+ * Returns price in BigNumber and leave formatting up to the caller
+ */
+export function calculatePrice({
+  buyToken: { amount: buyAmount, decimals: buyDecimals = 0 },
```

Comment:
ubernit: I usually prefer to do the destructuring in the body of the function so it's not so much info in the function declaration.

```js
export function calculatePrice(param: CalculatePriceParams) {
   const {
     buyToken: { amount: buyAmount, decimals: buyDecimals = 0 },
     sellToken: { amount: sellAmount, decimals: sellDecimals = 0 },
    } = params
    ...
}
```

Feel free to ignore, maybe this is oppinionated

Reply:
Done. I also renamed `buyToken` and `sellToken` to `numerator` and `denominator`, respectively, to be more in line with price and less bound to token idea.